### ISSUE TITLE
Rootstore can end up in a non-published state.

### ISSUE THREAD

**Issue by oed** [3box/3box-js#788]
Currently the rootstore is only published to the address server in one place: https://github.com/3box/3box-js/blob/develop/src/3box.js#L83
This could cause issues if for some reason the rootstore ends up in a non-published state. We should try to always check if the rootstore is published and republish it if needed. 

---

Conversation from discord:

```
FloToday at 12:53 AM
hey there :slight_smile: I still have IPFS issues with my 3box profile: https://www.3box.io/0xfac0475b677b54f72682e0ea633ffa1088110dcf here are two of URLs creating the error messages: https://ipfs.3box.io/profile?address=0xfac0475b677b54f72682e0ea633ffa1088110dcf and https://beta.3box.io/address-server/odbAddress/0xfac0475b677b54f72682e0ea633ffa1088110dcf ...can I manually unlink and link my address again?!  anyone an idea how to fix this? i cannot save anything in my profile..
oed 3⃣Today at 9:45 AM
Hey @Flo it does look like your profile is not linked currently. Did you manually unlink it? Or when did this start to happen?
FloToday at 10:09 AM
@oed 3⃣  it was right from the beginning.. initially it was because I had an old DID scheme i guess (it was did:muport:QmNRFFbyB4jcVxB3DDuWhzzGqxepDipq5xkB81M9TuUxXQ) but then the link process started and I got a new DID with did:3:bafyreidmg7fq6yxhacbjtzoecymrdmhboe5qwpdd6f22aaiv7ti7xu4yqa I thought about manually unlinking the address but first I wanted to ask if there is another way :wink:
oed 3⃣Today at 11:34 AM
@Flo ok, I'm asking because your address is not linked right now. So it seems like it needs to be linked again. What happens when you try to log into 3box.io?
FloToday at 11:39 AM
it seems like login worked but console with the error says otherwise :smile:
also tried deleting website data but still it just asks for a signature and shows me the same DID
oed 3⃣Today at 11:42 AM
What error are you seeing in the console @Flo ?
FloToday at 11:45 AM

0xfac.. is my address
oed 3⃣Today at 11:45 AM
ok, yeah. What happens if you try to update your data in the edit profile section?
FloToday at 11:46 AM

these errors come up but
but e.g. saving my name seems to work in my browser
but it is not saved in ipfs
when I take a look in another browser
so my name update seems to be saved in the local app state but not externally

those four calls result in a 404 error when updating my profile
oed 3⃣Today at 11:50 AM
Thanks @Flo I think most of those errors are unrelated. I'm creating a bug report related to your issue.
FloToday at 11:51 AM
ok thanks :slight_smile: do you think I can manually trigger the link process again?
oed 3⃣Today at 11:53 AM
If you want to dive in you could use the example dapp in the 3box-js repo (you have to set it up locally). There you can call box.linkAddress in the console. However I'm not sure that it would totally solve your issue.
```

**Comment by ColdDevil** [comment]
In case somebody needs a quick fix for this one:

- clone the https://github.com/3box/3box-js repo
- npm ci
- npm run example:start
- open the website localhost:3000 in your browser
- in case you are using Brave, deactivate the shield/adblocker for this site (localhost)
- authenticate by clicking the button, confirm meta mask popups
- start the link process again by clicking the button
- sign the message in metamask
- try accessing your profile at https://ipfs.3box.io/profile?address=0xYOURADDRESS

**Comment by oed** [comment]
Thanks @ColdDevil. I assume this is with your existing account? Or can you replicate this with a new account as well?

**Comment by ColdDevil** [comment]
This was a fix for my existing account that somehow got stuck without a linked address. With a new account everything went fine!

**Comment by oed** [comment]
Ok, that's good :) 
I was hoping that you'd found a way to replicate the issue though so that we could have solved it :)

**Comment by oed** [comment]
I believe this issue should be mostly solved by #798. Putting this in staging until it's released.

**Comment by ColdDevil** [comment]
In case somebody has the same issue.. I had to manually unlink my address by executing `box.removeAddressLink("0xfac0475b677b54f72682e0ea633ffa1088110dcf");` in the console of the aforementioned demo repo (see https://github.com/3box/3box-js#Box+removeAddressLink). Afterwards it worked again :)
